docs/section-7/swerve-drive.md

```diff
@@ -0,0 +1,164 @@
+        if (difference > 0.25) return true;
+        else return false;
+    }
+    /**
+     * Converts the angle from radians to a percentage of encoder ticks. Credit to Team 100 for their code.
+     * 
+     * @param angle             The desired angle from -1.0 to 1.0
+     * @param encoderPosition   The position of the <i> quadrature </i> encoder on the turn motor controller.
+     * @param gearRatio        The gear ratio of the turn motor controller.
+     * @return The converted angle between -0.5 and 0.5.
+     */
+    public static double convertAngle(double angle, double encoderPosition, double gearRatio) {
+        double encPos = encoderPosition / gearRatio;
+
+        double temp = angle;
+        temp += (int)encPos;
+
+        encPos = encPos % 1;
+
+        if ((angle - encPos) > 0.5) temp -= 1;
+        if ((angle - encPos) < -0.5) temp += 1;
+
+        return temp;
+    }
+```
+
+`computeSetpoints()` uses the other two functions in order to computer the desired speed as a percentage of maximum applied voltage (from -1.0 to 1.0) and the angle as a percentage of a full rotation of the encoder on the turn motor. `shouldReverse()` determines whether or not the module should turn to the desired angle (\(\alpha\)) or if it should turn to \(180^{\circ} - \alpha\) and run the drive motor in the opposite direction (which has the same result as turning to \(\alpha\) but may be faster). `convertAngle()` converts the angle from a fraction of a full revolution in radians to a fraction of a full revolution in quadrature encoder ticks.
+
+We can combine these three methods in our `move()` function as follows:
+
+```
+public void move(double normalizedSpeed, double angle) {
+    double setpoints[] = SwerveMath.computeSetpoints(normalizedSpeed / maxSpeed,
+                                                     angle / (2 * Math.PI),
+                                                     turn.getSelectedSensorPosition(0),
+                                                     gearRatio);
+    setSpeed(setpoints[0]);
+    if(setpoints[0] != 0.0) setAngle(setpoints[1]);
+}
+```
+
+## Home Absolute
+Currently, out `move()` method requires that our swerve modules start in a known configuration (let's assume straight forward) so th at passing and angle of 0 makes it face forward, 0.5 makes it face backward, etc. This is where the `HomeAbsolute` command comes in. `HomeAbsolute` requires knowing two things: the quadrature/analog position of the initial configuration and the gear ratio of the turn motor. We calculate our displacement from the configuration we want and use set the sensor position to this displacement. As a result, whenever we read from this sensor (or direct a motor controller to go to a specific position), the sensor value will be relative to the intial configuration. Here is the bulk of the command for Talon_SRX motor controllers:
```

Comment:
Would the code work properly if the known starting configuration was something other than straight forward?

"th at" -> "that"

"and angle" -> "an angle"

"use set" -> "set"

Reply:
If all of the modules were facing straight backward, wouldn't an angle of 0 correspond with straight backward and 0.5 as straight forward? It's my understanding that the code would work but the angles would be rotated by 180 degrees, which is not what we want.

Reply:
Yeah, in the situation you describe, the robot would drive as though it was turned 180. My point was "start in a known configuration (let's assume straight forward)" should really be "measure their orientation as a counter-clockwise angle relative to facing straight forward"